sdk/lib/opentelemetry/sdk/trace/tracer.rb

```diff
@@ -30,13 +30,14 @@ def start_root_span(name, attributes: nil, links: nil, start_timestamp: nil, kin
         def start_span(name, with_parent: nil, attributes: nil, links: nil, start_timestamp: nil, kind: nil)
           name ||= 'empty'
 
-          with_parent ||= Context.current
-          parent_span_context = OpenTelemetry::Trace.current_span(with_parent).context
+          parent_context = with_parent || Context.current
+          parent_span = OpenTelemetry::Trace.current_span(parent_context)
+          parent_span_context = parent_span.context
           if parent_span_context.valid?
             parent_span_id = parent_span_context.span_id
             trace_id = parent_span_context.trace_id
           end
-          @tracer_provider.internal_create_span(name, kind, trace_id, parent_span_id, attributes, links, start_timestamp, with_parent, @instrumentation_library)
+          @tracer_provider.internal_create_span(name, kind, trace_id, parent_span_id, attributes, links, start_timestamp, parent_context, parent_span, @instrumentation_library)
```

Comment:
Since this now passes in the parent span, does this function need all of the other parent related attributes like `parent_span_id` that could be queried on the `parent_span` object?

Reply:
That's fair - I'll move more of this into `internal_create_span` - the current code shape is the result of a merge without much thought.